rest/data/pod.go

```diff
@@ -42,3 +83,48 @@ func (c *MockPodConnector) CheckPodSecret(id, secret string) error {
 	}
 	return errors.New("pod does not exist")
 }
+
+func validatePod(p restModel.APICreatePod) (*pod.Pod, error) {
+	if p.Image == nil {
+		return nil, gimlet.ErrorResponse{
+			StatusCode: http.StatusBadRequest,
+			Message:    fmt.Sprintf("Invalid API input: missing image"),
+		}
+	}
+
+	for _, envVar := range p.EnvVars {
+		if envVar.SecretOpts == nil {
```

Comment:
You could simplify the nested ifs to if envVar.SecretOpts == nil && envVar.name == nil.